pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java

```diff
@@ -1004,6 +1004,59 @@ public void testTopicReplicatedAndProducerCreate(String topicPrefix, String topi
         nonPersistentProducer2.close();
     }
 
+    @Test
+    public void createPartitionedTopicTest() throws Exception {
+        final String cluster1 = pulsar1.getConfig().getClusterName();
+        final String cluster2 = pulsar2.getConfig().getClusterName();
+        final String cluster3 = pulsar3.getConfig().getClusterName();
+        final String namespace = randomSuffixString("pulsar/ns", 5);
+
+        final String persistentPartitionedTopic =
+                randomSuffixString("persistent://" + namespace + "/partitioned", 5);
+        final String persistentNonPartitionedTopic =
+                randomSuffixString("persistent://" + namespace + "/non-partitioned", 5);
+        final String nonPersistentPartitionedTopic =
+                randomSuffixString("non-persistent://" + namespace + "/partitioned", 5);
+        final String nonPersistentNonPartitionedTopic =
+                randomSuffixString("non-persistent://" + namespace + "/non-partitioned", 5);
+        final int numPartitions = 3;
+
+        admin1.namespaces().createNamespace(namespace, Sets.newHashSet(cluster1, cluster2, cluster3));
+        admin1.namespaces().setNamespaceReplicationClusters(namespace, Sets.newHashSet("r1", "r2", "r3"));
+
+        admin1.topics().createPartitionedTopic(persistentPartitionedTopic, numPartitions);
+        admin1.topics().createPartitionedTopic(nonPersistentPartitionedTopic, numPartitions);
+        admin1.topics().createNonPartitionedTopic(persistentNonPartitionedTopic);
+        admin1.topics().createNonPartitionedTopic(nonPersistentNonPartitionedTopic);
+
+        List<String> partitionedTopicList = admin1.topics().getPartitionedTopicList(namespace);
+        Assert.assertTrue(partitionedTopicList.contains(persistentPartitionedTopic));
+        Assert.assertTrue(partitionedTopicList.contains(nonPersistentPartitionedTopic));
+
+        // wait non-partitioned topics replicators created finished
+        Thread.sleep(1000);
```

Comment:
Flaky test is easy to appear in this way, can we use Awaitility?
`admin.topics().getList().size > x` can be used as the judgment condition

Reply:
Ok, I'll fix this. Thanks.